knimin/tests/test_ag_edit_barcode.py

```diff
@@ -89,6 +93,105 @@ def test_post(self):
+        # saving properly.
+        self.mock_login_admin()
+
+        # this is a barcode that belongs to a set of ag_kits which has multiple
+        # participant_names, i.e. sources
+        # SELECT barcode
+        # FROM barcodes.barcode
+        # JOIN ag.ag_kit_barcodes USING (barcode)
+        # JOIN ag.ag_kit USING (ag_kit_id)
+        # WHERE ag_login_id IN
+        #       (SELECT ag_login_id
+        #               FROM (SELECT array_agg(participant_name) as sources,
+        #                            count(participant_name) as numsources,
+        #                            ag_login_id FROM ag.ag_login_surveys
+        #                     GROUP BY ag_login_id) as foo
+        #               WHERE numsources > 2);
+        barcode = '000023125'  # current survey_id = "a4f1061f5bac9ae3"
+        details = db.getAGBarcodeDetails(barcode)
+        payload = {'barcode': barcode,
+                   'ag_kit_id': details['ag_kit_id'],
+                   'site_sampled': details['site_sampled'],
+                   'sample_date': details['sample_date'],
+                   'sample_time': details['sample_time'],
+                   'participant_name': details['participant_name'],
+                   'notes': details['notes'],
+                   'environment_sampled': details['environment_sampled'],
+                   'refunded': details['refunded'] or 'N',
+                   'withdrawn': details['withdrawn'] or 'N'}
+        response = self.post('/ag_edit_barcode/', payload)
+        self.assertEqual(response.code, 200)
+        # check that no actual change has happened
+
+        dbinfo = db.getAGBarcodeDetails(barcode)
+        for field in set(payload.keys()) - set(self.none_fields):
+            self.assertEqual(details[field], dbinfo[field])
+
+        # obtain all participant_names
+        sql = """SELECT DISTINCT participant_name
+                 FROM ag.ag_kit_barcodes
+                 LEFT JOIN ag.ag_kit USING (ag_kit_id)
+                 LEFT JOIN ag.ag_login_surveys USING (ag_login_id)
+                 WHERE barcode = %s"""
+        sourcenames = db._con.execute_fetchall(sql, [barcode])
+        self.assertTrue(sourcenames is not None)
+
+        sourcenames = [x[0] for x in sourcenames]
+        # changing source for the barcode
+        old_sourcename = payload['participant_name']
+        new_sourcename = list(set(sourcenames) -
+                              set(old_sourcename))[0]
+        payload['participant_name'] = new_sourcename
+
+        response = self.post('/ag_edit_barcode/', payload)
+        obs_details = db.getAGBarcodeDetails(barcode)
+
+        # revert to old participant_name to leave a clean DB
+        payload['participant_name'] = old_sourcename
+        response = self.post('/ag_edit_barcode/', payload)
+
+        self.assertEqual(response.code, 200)
+        self.assertEqual(new_sourcename, obs_details['participant_name'])
```

Comment:
change to `self.assertEqual(obs_details['participant_name'], new_sourcename)`
It is expected that the first parameter is the observed value and the second one is the expected one.